horovod/run/elastic/registration.py

```diff
@@ -100,7 +104,8 @@ def _record_state(self, host, slot, state):
                 logging.info('record state: {}[{}] = {}'.format(host, slot, state))
                 self._states[key] = state
                 self._workers[state].add(key)
-                rendezvous_id = self._rendezvous_id
+
+            rendezvous_id = self._rendezvous_id
```

Comment:
@EnricoMi note that this is a bugfix introduced in #1966.  Not sure if it's related to some of the errors we've been seeing, but worth being aware of.